bertopic/_save_utils.py

```diff
@@ -0,0 +1,315 @@
+        try:
+            get_hf_file_metadata(hf_hub_url(repo_id=repo_id, filename="README.md", revision=revision))
+        except:
+            model_card = model_card or {}
+            model_name = repo_id.split('/')[-1]
+
+            readme_text = generate_readme(model_card, model_name)
+            readme_path = Path(tmpdir) / "README.md"
+            readme_path.write_text(readme_text)
+
+        # Upload model
+        return upload_folder(repo_id=repo_id, folder_path=tmpdir, revision=revision,
+                             create_pr=create_pr, commit_message=commit_message)
+
+
+def load_local_files(path):
+    """ Load local BERTopic files """
+    # Load json configs
+    topics = load_cfg_from_json(path / TOPICS_NAME)
+    params = load_cfg_from_json(path / CONFIG_NAME)
+
+    # Load Topic Embeddings
+    safetensor_path = path / HF_SAFE_WEIGHTS_NAME
+    if safetensor_path.is_file():
+        tensors = safetensors.torch.load_file(safetensor_path, device="cpu")
+    else:
+        torch_path = path / HF_WEIGHTS_NAME
+        if torch_path.is_file():
+            tensors = torch.load(torch_path, map_location="cpu")
+
+    # c-TF-IDF
+    ctfidf_tensors = None
+    safetensor_path = path / CTFIDF_SAFE_WEIGHTS_NAME
+    if safetensor_path.is_file():
+        ctfidf_tensors = safetensors.torch.load_file(safetensor_path, device="cpu")
+    else:
+        torch_path = path / CTFIDF_WEIGHTS_NAME
+        if torch_path.is_file():
+            ctfidf_tensors = torch.load(torch_path, map_location="cpu")
+    ctfidf_config = load_cfg_from_json(path / CTFIDF_CFG_NAME)
+
+    return topics, params, tensors, ctfidf_tensors, ctfidf_config
+
+
+def load_files_from_hf(path):
+    """ Load files from HuggingFace. """
+    path = str(path)
+
+    # Configs
+    topics = load_cfg_from_json(hf_hub_download(path, TOPICS_NAME, revision=None))
+    params = load_cfg_from_json(hf_hub_download(path, CONFIG_NAME, revision=None))
+
+    # Topic Embeddings
+    try:
+        tensors = hf_hub_download(path, HF_SAFE_WEIGHTS_NAME, revision=None)
+        tensors = safetensors.torch.load_file(tensors, device="cpu")
+    except:
+        tensors = hf_hub_download(path, HF_WEIGHTS_NAME, revision=None)
+        tensors = torch.load(tensors, map_location="cpu")
+
+    # c-TF-IDF
+    try:
+        ctfidf_config = load_cfg_from_json(hf_hub_download(path, CTFIDF_CFG_NAME, revision=None))
+        try:
+            ctfidf_tensors = hf_hub_download(path, CTFIDF_SAFE_WEIGHTS_NAME, revision=None)
+            ctfidf_tensors = safetensors.torch.load_file(ctfidf_tensors, device="cpu")
+        except:
+            ctfidf_tensors = hf_hub_download(path, CTFIDF_WEIGHTS_NAME, revision=None)
+            ctfidf_tensors = torch.load(ctfidf_tensors, map_location="cpu")
+    except:
+        ctfidf_config, ctfidf_tensors = None, None
+
+    return topics, params, tensors, ctfidf_tensors, ctfidf_config
+
+
+def generate_readme(model_card: dict, model_name: str):
```

Comment:
For this, it might be worth exploring https://huggingface.co/docs/huggingface_hub/guides/model-cards at some point. There are a few possible enhancements we could make there:

- Trying to autogenerate some parts of the model card (for example, the Transformers Trainer API adds hyperparameters to a model card https://huggingface.co/biglam/detr-resnet-50_fine_tuned_nls_chapbooks#training-hyperparameters. I could imagine that things like the number of topics, number of training docs etc., could all be of interest. 

- Creating a template model card, i.e. something like (https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/templates/modelcard_template.md). This can be helpful as a starting point for a model card structure even if most sections are not completed. 

I think this can all come later, and happy to help with this part if you think it might interest you.

Reply:
Oh nice, that's perfect! That would make the process rather straightforward. Just to be sure I got this correctly. I would only need to:
* Create a template
* Create a model card from that template
* Extract the text from the model card
* and finally write that to `.md` the same way I did before. 

I want to make sure that when I run `upload_folder`, the model card is not a separate commit. 

All in all, I do not think it would be too much work, so I think I will simply add it to this PR. That way, there is little left for saving/loading a BERTopic model.

Reply:
Ah, it seems that to use `.from_template` an installation of `Jinja2` is a requirement whilst it is not a requirement of BERTopic.  Instead, I will likely use f-strings, albeit a bit hackier, to generate the template.

Reply:
> Ah, it seems that to use .from_template an installation of Jinja2 is a requirement whilst it is not a requirement of BERTopic. Instead, I will likely use f-strings, albeit a bit hackier, to generate the template.

Sorry for the slow reply, yes I think in this case to avoid a dependency f-strings probably make the most sense. Let me know if you need any help/input with this.

Reply:
Great, I finished up the last changes. I am currently working on support for both multi-modal (cluster images and/or text) and multi-aspect topic modeling (multiple representations simultaneously, keywords, summaries, sentiment, etc.). I intend to also save those aspects but since they are mostly just values/text that can be put in the `topics.json`, that process should be straightforward. 

So I believe the best thing to do may be to merge this PR and then continue with the multi-modal/multi-aspect topic modeling before slightly updating the saving procedure.

Reply:
That sounds awesome! Super cool to be adding multi-modal support for this (it gives me a few more use cases to use this!!). 

Let me know if you want me to review any of these changes or input anywhere else.

Reply:
I'll go ahead and merge this now so that I can continue with the multi-modal support

> Let me know if you want me to review any of these changes or input anywhere else.

Great, thanks! Most likely, I will ping you when I start saving the multi-modal model as I am not quite sure what the best way to save those images is, but I will let you know by then when I have something you can look at.

Reply:
Thanks, @MaartenGr, and really excited about all of this work!